packages/template-retail-react-app/app/hooks/use-datacloud.js

```diff
@@ -0,0 +1,481 @@
+                    personalizationId: recommenderDetails.recommenderName, //* The identifier of the personalization (e.g., recommendation), provided by the personalization service provider, that led to the event.
+                    personalizationContextId: recommenderDetails.__recoUUID //* The identifier, provided by the personalization service provider, of the specific content (e.g., product) associated with this event.
+                }
+            )
+        })
+
+        const identityProfile = this._concatenateEvents(
+            baseEvent,
+            this._generateEventDetails('identity', 'Profile'),
+            {
+                isAnonymous: args.isGuest,
+                firstName: args.firstName,
+                lastName: args.lastName
+            }
+        )
+
+        let contactPointEmail = null
+        if (args.email) {
+            contactPointEmail = this._concatenateEvents(
+                baseEvent,
+                this._generateEventDetails('contactPointEmail', 'Profile'),
+                {
+                    email: args.email
+                }
+            )
+        }
+
+        const interaction = {
+            events: [
+                identityProfile,
+                ...(contactPointEmail ? [contactPointEmail] : []),
+                ...catalogObjects
+            ]
+        }
+
+        try {
+            this.sdk.webEventsAppSourceIdPost(interaction)
+        } catch (err) {
+            logger.error('Error sending DataCloud event')
+        }
+    }
+}
+
+/**
+ * Custom hook for sending PWA Kit events to Data Cloud.
+ *
+ * This hook provides methods to track various user interactions, such as
+ * page views, product views, category views, search impressions, and recommendations.
+ *
+ * @returns {object} An object containing methods for sending different Data Cloud events
+ */
+const useDataCloud = () => {
+    const {getUsidWhenReady} = useUsid()
+    const {isRegistered} = useCustomerType()
+    const customerId = useCustomerId() // Bug PWA -> Needs converted to Promise
+    const {data: customer} = useCurrentCustomer()
+    const {site} = useMultiSite()
+    let {effectiveDnt} = useDNT()
+    effectiveDnt = false // Remove after demo
+
+    const getEventUserParameters = async () => {
+        return {
+            isGuest: isRegistered ? 0 : 1,
+            usid: await getUsidWhenReady(),
+            customerId: customerId,
+            customerNo: customer?.customerNo,
+            firstName: customer?.firstName,
+            lastName: customer?.lastName,
+            email: customer?.email
+        }
+    }
+
+    const {
+        app: {dataCloudAPI: config, defaultSite: siteId}
+    } = getConfig()
+
+    const {appSourceId, tenantId} = config
+    let sdk = null
+    let isDatacloudInitiated = true
+
+    if (!appSourceId || !tenantId) {
+        isDatacloudInitiated = false
+    } else {
+        sdk = initDataCloudSdk(tenantId, appSourceId)
```

Comment:
Why did the initiation here instead of in the dataCloud Class? It is likely we only use one instance of DataClass and one instance of the Data Cloud SDK in entire app. Can we move it there?
Besides, this block code is likely to be run every time the hook is re-rendered, which is not a good practise for to initiate a sdk/class.